### PR TITLE
feat(pci): force default payment choice using ng-ovh-payment-method-register

### PR DESCRIPTION
**Pull request by y4nnL**
| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #TAPC-376
| License          | BSD 3-Clause

- [X] Try to keep pull requests small so they can be easily reviewed.
- [X] Commits are signed-off
- [X] Only FR translations have been updated
- [X] Branch is up-to-date with target branch
- [X] Lint has passed locally
- [X] Standalone app was ran and tested locally
- [X] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

### CODE REVIEW

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js [hunk 1]
```diff
@@ -48,4 +58,23 @@ export default class OvhPaymentMethodIntegrationInContextPaypalCtrl {
...
+    if (this.inContextCtrl.integrationCtrl.disabled) {
+      this.disableButtons();
+    }
+  }
+
+  disableButtons() {
+    if (this.buttonsCtrl) {
+      this.buttonsCtrl.disable();
+    }
+  }
```
Comment (darsene) [inline packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js:73]: To reduce code, you can group enable and disable state in one function
```suggestion
  manageButtonsState(disabled = true) {
    if (this.buttonsCtrl) {
      disabled ? this.buttonsCtrl.disable() :  this.buttonsCtrl.enable() ;
    }
  }
```

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js [hunk 2]
```diff
@@ -48,4 +58,23 @@ export default class OvhPaymentMethodIntegrationInContextPaypalCtrl {
...
+
+  enableButtons() {
+    if (this.buttonsCtrl) {
+      this.buttonsCtrl.enable();
+    }
+  }
```
Comment (darsene) [inline packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js:79]:
```suggestion
```

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js [hunk 3]
```diff
@@ -48,4 +58,23 @@ export default class OvhPaymentMethodIntegrationInContextPaypalCtrl {
       .then(({ formSessionId }) => formSessionId)
       .catch(() => {});
   }
+
+  onButtonsInit(_, buttonsCtrl) {
+    this.buttonsCtrl = buttonsCtrl;
+    if (this.inContextCtrl.integrationCtrl.disabled) {
+      this.disableButtons();
```
Comment (darsene) [inline packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js:65]:
```suggestion
      this.manageButtonsState();
```

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js [hunk 4]
```diff
@@ -2,9 +2,19 @@ import { get, merge } from 'lodash-es';
...
     this.$timeout = $timeout;
     this.ovhPaymentMethod = ovhPaymentMethod;
+    this.buttonsCtrl = null;
+
+    $scope.$watch(
+      () => this.inContextCtrl.integrationCtrl.disabled,
+      (isIntegrationDisabled) => {
+        this.disabled = isIntegrationDisabled;
+        if (this.disabled) this.disableButtons();
+        else this.enableButtons();
```
Comment (darsene) [inline packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js:15]:
```suggestion
       this.manageButtonsState(isIntegrationDisabled);
```

**Review by antonymarion — DISMISSED** [review]

File: packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js (hunk 1 above)
Comment (y4nnL) [reply packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js:73]: I disagree, this lacks readability, you have to look into `manageButtonsState` and pay attention to the `disabled` argument to know what the function does. To me it's not worth it just for a few lines of code. In the other hands `disableButtons` and `enableButtons` are pretty self explanatory

Comment (darsene) [reply packages/components/ng-ovh-payment-method/src/components/integration/inContext/paypal/controller.js:73]: Well, it was just an example but you can rename it for more readability 
```suggestion
  activateButtons(disabled) {
         if (this.buttonsCtrl) {
      disabled ? this.buttonsCtrl.disable() :  this.buttonsCtrl.enable() ;
    }
    }
  }
```

**Review by darsene — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**